lib/stoplight/data_store/redis.rb

```diff
@@ -40,10 +40,23 @@ def get_failures(light)
         normalize_failures(query_failures(light), light.error_notifier)
       end
 
+      WINDOW_SIZE = Float::INFINITY
+
+      #  TODO: time in utc
+      #  TODO: investigate warnings
+      #  TODO: tests with different window sizes
       def record_failure(light, failure)
-        size, = @redis.multi do
-          @redis.lpush(failures_key(light), failure.to_json)
-          @redis.ltrim(failures_key(light), 0, light.threshold - 1)
+        # @redis.lpush(failures_key(light), failure.to_json)
+        # @redis.ltrim(failures_key(light), 0, light.threshold - 1)
+        failures_key = failures_key(light)
+
+        _, size, = @redis.multi do
+          @redis.zadd(failures_key, failure.time.to_i, failure.to_json)
+          @redis.zcard(failures_key)
+          # keep at most WINDOW_SIZE old errors
+          @redis.zremrangebyscore(failures_key, 0, Time.new.to_i - WINDOW_SIZE)
```

Comment:
This is awesome.. Can I help you with this, this is super useful to me, and I need it asap!. thanks!